### ISSUE TITLE
pass data from modal when hiding

### ISSUE THREAD

**Issue by abdulqadir93** [mseemann/angular2-mdl#263]
I'm using `showCustomDialog` mostly through-out my application. I've hit some use-cases where the dialog just needs to pass some data when closed. Passing a provider for this sometimes ... well .. seems over-work. I think it would be ideal to be able to pass some data to the MdlDialogReference's `hide` method which MdlDialogReference's `onHide` could optionally receive.

```
  this.dialogService.showCustomDialog({ ... })
    .subscribe(dialogRef => {
       dialogRef.onHide().subscribe(data => {  
          // do whatever with the data!
       });
    });
```

I'll be very much happy to issue a pull-request if you think that's a valid use-case. Please let me know your thoughts.

**Comment by mseemann** [comment]
hi. yes please provide a pr for this feature. thx!

**Comment by mseemann** [comment]
published in 2.6.0
